packages/iov-ethereum/src/ethereumcodec.ts

```diff
@@ -0,0 +1,51 @@
+// import createKeccakHash from "keccak";
+
+import {
+  Address,
+  Nonce,
+  SignedTransaction,
+  SigningJob,
+  TransactionIdBytes,
+  TxCodec,
+  UnsignedTransaction,
+} from "@iov/bcp-types";
+import { Keccak256 } from "@iov/crypto";
+import { Encoding } from "@iov/encoding";
+import { ChainId, PostableBytes, PublicKeyBundle } from "@iov/tendermint-types";
+
+const { toHex } = Encoding;
+
+export function toChecksumAddress(address: string): Address {
+  // https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md
+  const addressLower = address.toLowerCase().replace("0x", "") as Address;
+  const addressHash = toHex(new Keccak256(addressLower).digest());
+  let checksumAddress = "0x";
+  for (let i = 0; i < 40; i++) {
+    checksumAddress += parseInt(addressHash[i], 16) > 7 ? addressLower[i].toUpperCase() : addressLower[i];
+  }
+  return checksumAddress as Address;
+}
+
+export const ethereumCodec: TxCodec = {
+  bytesToSign: (unsigned: UnsignedTransaction, nonce: Nonce): SigningJob => {
+    throw new Error(`Not implemented utx: ${unsigned}, nonce: ${nonce}`);
+  },
+  bytesToPost: (signed: SignedTransaction): PostableBytes => {
+    throw new Error(`Not implemented tx: ${signed}`);
+  },
+  identifier: (signed: SignedTransaction): TransactionIdBytes => {
+    throw new Error(`Not implemented tx: ${signed}`);
+  },
+  parseBytes: (bytes: PostableBytes, chainId: ChainId): SignedTransaction => {
+    throw new Error(`Not implemented bytes: ${bytes}, chainId: ${chainId}`);
+  },
+  keyToAddress: (pubkey: PublicKeyBundle): Address => {
+    const hash = toHex(new Keccak256(pubkey.data).digest());
```

Comment:
Could you add two checks here:

* `pubkey.algo === Algorithm.Secp256k1`
* `pubkey.data.length === 64`

In both cases just trow an exception.

Reply:
The function should expect uncompressed pubkeys with prefix byte `04`, so you need to hash `toHex(new Keccak256(pubkey.data.slice(1)).digest());` and add "04" to the test